xdp/src/program.rs

```diff
@@ -0,0 +1,195 @@
+#![allow(clippy::arithmetic_side_effects)]
+
+#[cfg(target_os = "linux")]
+use aya::{programs::Xdp, Ebpf};
+use std::io::{Cursor, Write};
+
+macro_rules! write_fields {
+    ($w:expr, $($x:expr),*) => {
+        $(
+            $w.write_all(&$x.to_le_bytes())?;
+        )*
+    };
+}
+
+const SHT_NULL: u32 = 0;
+const SHT_PROGBITS: u32 = 1;
```

Comment:
I am assuming these constants are documented somewhere, can you add comments here and/or link to documentation in case someone needs to understand what this code does?

Reply:
I've added some comments, but obvs this stuff is in the ELF spec. Very googleable :P

Reply:
(Also, again, this code is going away as soon as I'm happy with the API and can do a new aya release)

Reply:
There is nothing more permanent then temporary solutions:) Thank you!